packages/flutter/lib/src/cupertino/nav_bar.dart

```diff
@@ -1163,6 +1165,14 @@ class _CupertinoSliverNavigationBarState extends State<CupertinoSliverNavigation
   @override
   void didChangeDependencies() {
     super.didChangeDependencies();
+    isLarge = MediaQuery.orientationOf(context) != Orientation.landscape;
```

Comment:
Extreme nit: `isLarge` might be confusing. Whether it's large or not is usually determined by the constructor or scroll, while this is only concerned with the orientation. `isPortrait` or `canBeLarge` would probably be better.